x/wasm/internal/keeper/ioutil_test.go

```diff
@@ -57,31 +60,38 @@ func TestUncompress(t *testing.T) {
 			src:      wasmGzipped[:len(wasmGzipped)-5],
 			expError: io.ErrUnexpectedEOF,
 		},
+		"handle limit gzip output": {
```

Comment:
Huh, this passes? That means it can return `(n, EOF)` where `n > 0`?

The test cases (both limit and limit+1 gzips) pass. So I can approve it. Just seems there is some possible way something can slip through.